api/src/wfl/module/batch.clj

```diff
@@ -63,10 +63,22 @@
 
 (defn update-workload!
   "Use transaction TX to batch-update WORKLOAD statuses."
+  [tx workload]
+  (letfn [(update! [{:keys [id] :as workload}]
+            (postgres/batch-update-workflow-statuses! tx workload)
+            (postgres/update-workload-status! tx workload)
+            (workloads/load-workload-for-id tx id))]
+    (util/when-> workload #(and (:started %) (not (:finished %))) update!)))
+
+(defn stop-workload!
+  "Use transaction TX to stop the WORKLOAD."
   [tx {:keys [id] :as workload}]
-  (if (or (:finished workload) (not (:started workload)))
-    workload
-    (do
-      (postgres/batch-update-workflow-statuses! tx workload)
-      (postgres/update-workload-status! tx workload)
-      (workloads/load-workload-for-id tx id))))
+  (letfn [(patch-workload! [cols]
+            (jdbc/update! tx :workload cols ["id = ?" id]))
+          (stop! [{:keys [id] :as workload}]
+            (let [now (OffsetDateTime/now)]
+              (when-not (:started workload)
+                (patch-workload! {:started now :finished now}))
```

Comment:
interesting, so the rationale is that a stopped workload must be a "finished" one?

Reply:
Nope. `:stopped` is a state somewhere between `:started` and `:finished`. If a workload wasn't `:started`, when you stop it, it's a no op to transition to `:finsihed` so I just did it here.
Also I'm not sure the update code that talks to terra/cromwell will handle :started` workloads that haven't actually had any workflows submitted.